pkg/apis/folder/v0alpha1/register.go

```diff
@@ -1,10 +1,9 @@
 package v0alpha1
 
 import (
+	grafanaregistry "github.com/grafana/grafana/pkg/apiserver/registry/generic"
```

Comment:
this change does not feel right -- IIUC, now all the types depend on apiserver, not just apimachinery 🤔

Reply:
i guess it depends on where the `ResourceInfo` is built. it's only used by the api registry, correct?

Reply:
maybe -- IIUC, a client could also call `newObject()` and does not need the apiserver.  But I think we will likely have typed clients so maybe a non-issue.

Regardless I don't see why the key function needs to be on each kind -- it is the same all the time, right?

Reply:
> I don't see why the key function needs to be on each kind 

mostly because it seemed like the use case matched the other usages of ResourceInfo. i can rework it so it's not a method

Reply:
updated in https://github.com/grafana/grafana/pull/89635/commits/ec935986be833cf71819ae871d82f4d2e3c56a66